contracts/ResilientOracle.sol

```diff
@@ -10,6 +10,13 @@ import "./interfaces/OracleInterface.sol";
 contract ResilientOracle is OwnableUpgradeable, PausableUpgradeable, ResilientOracleInterface {
     uint256 public constant INVALID_PRICE = 0;
 
+    /// @notice vBNB address
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable vBnb;
+
+    /// @notice Set this as asset address for BNB. This is the underlying for vBNB
+    address public constant BNB_ADDR = 0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB;
```

Comment:
If this is hardcoded, how will it get updated when we deploy to different networks / chains?

Reply:
Should this also be set as immutable and set in the constructor?

Reply:
This is actually a good point as we are kinda hardcoding the behavior of the virtual native token as "BNB". I think for different networks we will have to make the oracle logic a bit more generic to make it network-independent. Currently, every price fetch is dependant on BNB.

Reply:
This is kinda network independent as the BNB_ADDR can denote the native token of any network.